Optimize root domain search through api

keryfan opened the pull request.

keryfan commented:
I have read the guide. This PR just adds filtering to root-domains check to speed-up the process, the rest of the code is same.

Neilpang commented:
please follow the dns api test:
https://github.com/acmesh-official/acme.sh/wiki/DNS-API-Test

Any changes to the dns api hook must be verified by the dns api test.

keryfan commented:
> please follow the dns api test: https://github.com/acmesh-official/acme.sh/wiki/DNS-API-Test
> 
> Any changes to the dns api hook must be verified by the dns api test.

Please check again - I've updated settings for actions and added corresponding secrets to repo. Then triggered activity by syncing latest changes from repo to fork.

Neilpang commented:
it's not passing 

https://github.com/fornex-com/acme.sh/actions/runs/14908908134

keryfan commented:
> it's not passing
> 
> https://github.com/fornex-com/acme.sh/actions/runs/14908908134

For some reason it fails on the process of creating container for **clearlinux:latest**, even before starting cert request.
![image](https://github.com/user-attachments/assets/d2ca5a6e-7ae0-418f-a69f-120e37c1d8f0)

But at the same time all tests with different docker images were successful.

I also tried to run acmetest locally and also got error when testing only clearlinux platform on the phase of creation of container itself.

![image](https://github.com/user-attachments/assets/de1d825e-5ae7-4494-adb1-13b98ff9db1b)

Neilpang commented:
please run it again, it should work.

keryfan commented:
Please check https://github.com/fornex-com/acme.sh/actions/runs/14908908134. All actions completed.

keryfan commented:
Please check again, al checks have passed.
